src/ng/location.js

```diff
@@ -482,7 +482,7 @@ var locationPrototype = {
    * ```
    *
    * @param {(string|number)=} path New path
-   * @return {string} path
+   * @return {(string|object)} path if called with no parameters, or `$location` if called with a parameter
```

Comment:
I don't think objects are supported per se, they just work because we call toString on path, which was added here: https://github.com/angular/angular.js/issues/7054. I think we shouldn't have built in handling for intergers, as the docs clearly say the path must be a string.